hummingbird/ml/_container.py

```diff
@@ -67,6 +67,46 @@ def __init__(self, model, n_threads=None, batch_size=None, extra_config={}):
     def model(self):
         return self._model
 
+    def _run(self, function, *inputs, reshape=False):
+        """
+        This function either score the full dataset at once or triggers batch inference.
+        """
+
+        if self._batch_size is None:
+            return function(*inputs)
+        else:
+            return self._run_batch_inference(function, *inputs, reshape=reshape)
+
+    def _run_batch_inference(self, function, *inputs, reshape=False):
+        """
+        This function contains the code to run batched inference.
+        """
+
+        if DataFrame is not None and type(inputs[0]) == DataFrame:
+            # Split the dataframe into column ndarrays.
+            inputs = inputs[0]
```

Comment:
codecov claims line 87-91 aren't tested. Can you please either add a case for this, or dig into why codecov is confused?

Reply:
Ah I don't have a test for DataFrame! Let me add it.

Reply:
Ok it should be good now. It took a while to get it right, dataframe support is not solid at the moment.